overlord/snapstate/snapstate.go

```diff
@@ -471,6 +473,19 @@ func WaitRestart(task *state.Task, snapsup *SnapSetup) (err error) {
 		return fmt.Errorf("there was a snapd rollback across the restart")
 	}
 
+	// if we have restarted and snapd was refreshed, then we need to generate
+	// snapd wrappers again with current snapd, as the logic of generating
+	// wrappers may have changed between previous and new snapd code.
+	if snapsup.Type == snap.TypeSnapd && !release.OnClassic {
```

Comment:
I was a bit confused why this is here and not at the end, but then noticed that the code below is for base/kernel and will return nil early otherwise.  I think it might be time to reorg the code a bit, and have a unified stanza dedicated to snapd, and the a comment after about // consider kernel and base case.

```
if snapup.Type = snap.TypeSnapd {
   ... all snapd logic
}

// consider kernel and base
...
```